src/manager.ts

```diff
@@ -487,25 +489,37 @@ export class ThreadedClassManagerClassInternal extends EventEmitter {
 				// Close the child processes upon exit:
 				process.stdin.resume() // so the program will not close instantly
 
-				const exitHandler = (options: any, err: Error) => {
+				// Read about Node signals here:
+				// https://nodejs.org/api/process.html#process_signal_events
+
+				const onSignal = (signal: string, message?: string) => {
+					let msg = `ThreadedClass: Signal "${signal}" event`
+					if (message) msg += ', ' + message
+					console.log(msg)
+
 					this.killAllChildren()
 					.catch(console.log)
-					if (err) console.log(err.stack)
-					if (options.exit) process.exit()
+
+					process.exit()
 				}
 
-				// do something when app is closing
-				process.on('exit', exitHandler.bind(null, { cleanup: true }))
+				// Do something when app is closing:
+				process.on('exit', (code: number) => onSignal('process.exit', `exit code: ${code}`))
 
 				// catches ctrl+c event
-				process.on('SIGINT', exitHandler.bind(null, { exit: true }))
+				process.on('SIGINT', () => onSignal('SIGINT'))
```

Comment:
These handlers feel like bad citizens, as we will be forcing an exit even if the user tries to block exiting from these signals in other code.

Reply:
Notes after discussion with Julusian: Since these parts are opt-out with `this.dontHandleExit`, this should be ok.

Reply:
https://github.com/sindresorhus/exit-hook
Does a similar thing, and as noone has complained there, this seems like an acceptable practise for now